namer/core/src/main/scala/io/buoyant/namer/NameTreeTransformer.scala

```diff
@@ -49,6 +57,8 @@ trait NameTreeTransformer {
       }
```

Comment:
Would it make sense to do that instead to avoid the getSelf() ? 


```scala
  private def makeWithAdminHandlers(namer:Namer,nameTreeTransformer: NameTreeTransformer) =
    new Namer with Admin.WithHandlers with WithNameTreeTransformer {
      private[this] def isBound(tree: NameTree[Name]): Boolean = {
        tree match {
          case NameTree.Neg | NameTree.Empty | NameTree.Fail => true
          case NameTree.Alt(trees@_*) => trees.forall(isBound)
          case NameTree.Union(trees@_*) => trees.map(_.tree).forall(isBound)
          case NameTree.Leaf(_: Name.Bound) => true
          case NameTree.Leaf(_) => false
        }
      }

      override def lookup(path: Path): Activity[NameTree[Name]] = {
        namer.lookup(path).flatMap { tree =>
          if (isBound(tree))
            transform(tree.asInstanceOf[NameTree[Name.Bound]])
          else
            Activity.value(tree)
        }
      }

      override def transformer: NameTreeTransformer = nameTreeTransformer

      override def adminHandlers: Seq[Admin.Handler] = namer match {
        case withHandlers: Admin.WithHandlers => withHandlers.adminHandlers
        case _ => Nil
      }
    }
  
  def wrap(underlying: Namer): Namer = makeWithAdminHandlers(underlying, this)
```

Reply:
Not sure, getSelf seems clean enough to me. Mind that I've changed the code to have support for multiple transformers per namer/interpreter.

Reply:
Why would changing the code to have support for multiple transformers make any difference in this case ? You can just have 
`private def makeWithAdminHandlers(namer:Namer,nameTreeTransformers: Seq[NameTreeTransformer]) = ... `

Reply:
ok, no difference, I've changed the code to your suggestion.